libcontainer/factory_linux.go

```diff
@@ -239,19 +239,23 @@ func (l *LinuxFactory) StartInitialization() (err error) {
 		}
 		*pair.v = i
 	}
+
 	var (
 		pipe = os.NewFile(uintptr(pipefd), "pipe")
 		it   = initType(os.Getenv("_LIBCONTAINER_INITTYPE"))
 	)
+
+	// ensure that this pipe is always closed in the case of failure.
+	defer pipe.Close()
```

Comment:
Its still unneeded and putting it directly after the creation is both clearer and more idiomatic.

Here's a playground demo: https://play.golang.org/p/_KvjCwK2V4